openai-core/src/commonMain/kotlin/com.aallam.openai.api/chat/JsonData.kt

```diff
@@ -0,0 +1,39 @@
+package com.aallam.openai.api.chat
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonDecoder
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonEncoder
+import kotlin.jvm.JvmInline
+
+@JvmInline
+@Serializable(with = JsonData.JsonDataSerializer::class)
+public value class JsonData(public val jsonData: String){
```

Comment:
I was considering the idea of using `JsonObject` from kotlinx.serialization and exposing it as an `api` rather than as an `implementation` in Gradle. This approach could allow users to use the DSL to construct JSON schemas, and we might potentially add some helper functions later to simplify the process.

Reply:
I changed jsonData to be jsonElement and added some companion helper classes